src/ajax.js

```diff
@@ -597,15 +598,16 @@ jQuery.extend( {
 
 			// If data is available, append data to url
 			if ( s.data ) {
-				cacheURL += ( rquery.test( cacheURL ) ? "&" : "?" ) + s.data;
+				cacheURL = cacheURL.replace( remptyQuery, "" ) +
+					( rquery.test( cacheURL ) ? "&" : "?" ) + s.data;
```

Comment:
If we really want to update the "append parameter" logic to handle terminal `?`s, I'd rather see something like `(cacheURL + "&").replace( /\?(.+&$)|\??&$/, "?$1" )` (dropping `rquery` in favor of the new regex) in a separate PR also including [src/ajax/jsonp.js#L47](https://github.com/jquery/jquery/blob/305f193aa57014dc7d8fa0739a3fefd47166cd44/src/ajax/jsonp.js#L47).